client/src/components/AlumniDirectory.js

```diff
@@ -102,32 +115,78 @@ export default class AlumniDirectory extends Component {
     populateSearchDropdownStates(entries) {
         let gradYears = [];
         let allText = [];
+        let topicsArray = [];
+        let topicsSet = new Set();
+        let interestsArray = [];
+        let interestsSet = new Set();
         let display = [];
-        let i = 0;
+        let i = 0, j = 0;
+
         for (let post of entries) {
-            if(!gradYears.find(year => year['value'] === post.gradYear)) {
+            if (!gradYears.find(year => year['value'] === post.gradYear)) {
                 gradYears.push({
                     key: post.gradYear,
                     text: post.gradYear,
                     value: post.gradYear
                 });
             }
+
+            let topicsForAlumnus = post.topics;
+            for (let topic in topicsForAlumnus) {
+                if (!topicsSet.has(topic)) {
+                    topicsSet.add({
+                        key: post.topics[topic],
```

Comment:
Similarly these should be `key:  topicsForAlumnus[topic]` and so on... as the index topic does not relate to the post.topics array